deepmd/pt/model/descriptor/sezm_nn/triton/so2_rotation.py

```
@@ -0,0 +1,1726 @@
+
+
+def _canonical_block_index_values(coeff_index: Tensor, lmax: int) -> bool:
+    """Return whether real ``coeff_index`` values match the m-major mmax=1 layout."""
+    if getattr(coeff_index, "fake_mode", None) is not None:
+        return True
+    if coeff_index.device.type == "meta":
+        return True
+    try:
+        expected = build_m_major_index(int(lmax), 1, device=coeff_index.device)
+        return torch.equal(coeff_index, expected)
+    except Exception:  # pragma: no cover - exotic tensor subclasses
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)